### DISCUSSION TITLE
API endpoint documentation

### DISCUSSION THREAD

**Discussion by deanobaldwin** [amidaware/tacticalrmm#900] [General]
Hi, 

I was wondering if there was any supporting documentation around the APIs. I cannot seem to find much about it other than how to create an API key.

Many thanks,
Dean

**Comment by wh1te909** [comment]
just updated docs https://wh1te909.github.io/tacticalrmm/functions/api/#enable-swagger
let me know if that helps

**Reply by deanobaldwin** [reply]
Perfect

**Reply by deanobaldwin** [reply]
Just one last one to help me get my head around the code GetAddClients. I am trying to create a new client with site via the POST to /clients API endpoint. However, I receive the error:

"client with this name already exists."

even when creating a client that doesn't exist. Would you mind providing a brief example of a full create client/site json which will help me get my head around the schema.

**Reply by wh1te909** [reply]
edit: for adding clients

```json
{
    "client": {
        "name": "Client Name"
    },
    "site": {
        "name": "Site Name"
    },
    "custom_fields": []
}
```

**Reply by wh1te909** [reply]
for adding new site:
```json
{
    "site": {
        "client": 34, // primary key of the client
        "name": "NewSite123"
    },
    "custom_fields": []
}
```

**Reply by deanobaldwin** [reply]
Brilliant, thanks @wh1te909. Works perfectly.
